docs/howtos/port-linkml.md

```diff
@@ -0,0 +1,293 @@
+
+For a small but growing set of languages there is support for
+[generating data access object
+models](https://linkml.io/linkml/generators/index.html) for that
+language. For other languages it should be possible to code up
+generators using the flexible Jinja2 template based generator
+framework.
+
+However, there may still be cases where it is desirable to have full
+LinkML support within a language other than python, typically when
+writing a generic application that is designed to work with *any*
+LinkML model.
+
+Examples of these might be:
+
+1. A generic data editor or visualizer for LinkML models
+2. A code generator for a particular language where one wants to take advantage of features of that language
+
+For more on the first use case, see the [tool developer guide](https://linkml.io/linkml/developers/tool-developer-guide.html).
+
+As an example of the second, consider writing a code generator for
+Java that creates data access classes with the ability to customize
+the generation for various different styles of java annotations
+(Jackson, Lombok, Hibernate, ...). While this could be done in Python
+and Jinja2 templates, there is an advantage to leveraging existing
+tooling in the host language.
+
+It may be tempting to take the "easy" approach and simply write code
+that works directly from LinkML yaml files. This is a good approach if
+you are targeting one of the simple sub-profiles of LinkML but if you
+are aiming for a more powerful library then this has a number of disadvantages.
+
+## Definitions
+
+- LinkML framework: a general purpose programming library for working with LinkML models
+- Target Language: the programming language one wishes to write a LinkML framework for
+- LinkML metamodel: A schema that describes LinkML itself
+
+# Overview of Python Framework
+
+Currently there are 3 core repos:
+
+- [linkml-model](https://github.com/linkml/linkml-model) This holds the Source of Truth for the *metamodel*
+- [linkml-runtime](https://github.com/linkml/linkml-runtime) A runtime library that is depended on by schema-specific generated python dataclass libraries
+- [linkml](https://github.com/linkml/linkml) The core tooling, including [generators](https://linkml.io/linkml/generators/index.html)
+
+This guide assumes familiarity with all of the above
+
+Note that there are no runtime dependencies on linkml-model; this
+would create a circularity since the metamodel is like any other
+schema and depends on the runtime. Instead, linkml-model is synced into linkml-runtime.
+
+# How to Port to your language of choice
+
+## Step 1: Bootstrap metamodel data access classes
+
+Even if you intended to ultimately use the target language for doing
+codegen, we strongly recommend you bootstrap by using the existing
+generators framework. This may take a little bit of Python coding and some Jinja2 templates.
+
+Consult some of the existing generators to see how this is done. Be
+sure to use the newer style SchemaView generators.
```

Comment:
Add link to the directory, or a couple of examples of a `SchemaView` generator

Reply:
done.